runtime/v2/manager.go

```diff
@@ -33,13 +33,20 @@ import (
 	"github.com/containerd/containerd/platforms"
 	"github.com/containerd/containerd/plugin"
 	"github.com/containerd/containerd/runtime"
+	"github.com/containerd/containerd/runtime/v2/runc/options"
+	"github.com/containerd/typeurl"
+	ptypes "github.com/gogo/protobuf/types"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
 // Config for the v2 runtime
 type Config struct {
 	// Supported platforms
 	Platforms []string `toml:"platforms"`
+	// RuncRoot specifies the default root directory for runc when using the
+	// io.containerd.runc.v2 runtime.  When not set, the shim uses its own
+	// default value (which is specified in utils.RuncRoot).
+	RuncRoot string `toml:"runc_root"`
```

Comment:
The runtime/v2 pkg should remain decoupled from runc-specific concepts.

I’d rather prefer adding general-purpose annotations (map[string]string) to the plugin infra.

Reply:
> The runtime/v2 pkg should remain decoupled from runc-specific concepts.

That's fair.  I wasn't sure about the best way to pass this information in; the existing `Config` object seemed like the least-intrusive approach and then I named it `runc_root` because the `options.Options` is runc-specific.  I had initially looked at adding this to the tasks-service. but the manager looked like it was closer to where the value would need to take effect.  There's also a `runtime_root` configuration that exists for the CRI plugin if we had wanted a more-generic name here.

> I’d rather prefer adding general-purpose annotations (map[string]string) to the plugin infra.

I'm having a hard time visualizing what you're suggesting here.  Are you talking about changing `plugin.Registration` to include a `map[string]string`?

Reply:
> Are you talking about changing plugin.Registration to include a map[string]string?


Yes

Reply:
Another way is to support specifying in `[plugins."io.containerd.runc.v2i"]` in config.toml.  That might be better.